src/main.rs

```diff
@@ -89,15 +105,46 @@ fn main() {
 
 fn work(args: Args) -> anyhow::Result<()> {
     match args.command {
-        Some(Cmd::Set(Set { image, comp, .. })) if image.is_file() => match comp {
-            Compositor::Sway => swaybg(&image),
-            Compositor::X11 => hsetroot(&image),
+        Some(Cmd::Set(Set { output, image, comp, kill_swaybg, .. })) if image.is_file() => match comp {
+                Compositor::Sway => {
+                    if kill_swaybg {
+                        Command::new("pkill").arg("swaybg").output()?;
+                    }
+                    let mut res: anyhow::Result<()> = Ok(());
+                    for o in output {
+                        res = swaybg(o.as_str(), &image)
+                    }
+                    res
+                },
+                Compositor::X11 => {
+                    let mut res: anyhow::Result<()> = Ok(());
+                    for o in output {
+                        res = hsetroot(o.as_str(), &image)
+                    }
+                    res
+                },
         },
-        Some(Cmd::Random(Random { dir, comp, .. })) if dir.is_dir() => {
-            let p = rand_img(&dir)?;
+        Some(Cmd::Random(Random { output, dir, comp, kill_swaybg, .. })) if dir.is_dir() => {
             match comp {
-                Compositor::Sway => swaybg(&p),
-                Compositor::X11 => hsetroot(&p),
+                Compositor::Sway => {
+                    if kill_swaybg {
+                        Command::new("pkill").arg("swaybg").output()?;
+                    }
+                    let mut res: anyhow::Result<()> = Ok(());
+                    for o in output {
+                        let p = rand_img(&dir)?;
+                        res = swaybg(o.as_str(), &p)
+                    }
+                    res
+                },
+                Compositor::X11 => {
+                    let mut res: anyhow::Result<()> = Ok(());
+                    for o in output {
```

Comment:
Here as well. The possibility of the user having specified no `--output`s should be considered.

Reply:
I fixed this too.